packages/http-client-csharp/eng/scripts/RegenPreview-Local.ps1

```diff
@@ -0,0 +1,849 @@
+#Requires -Version 7.0
+
+<#
+.SYNOPSIS
+    Builds local generator packages and regenerates Azure SDK for .NET libraries for validation.
+
+.DESCRIPTION
+    This script:
+    1. Builds a local npm package of @typespec/http-client-csharp with a versioned name (1.0.0-alpha.YYYYMMDD.hash)
+    2. Builds and packages the three NuGet generator framework packages with the same versioning
+    3. Updates Packages.Data.props in azure-sdk-for-net with the local NuGet version
+    4. Updates the Azure generator (@azure-typespec/http-client-csharp) to use the local unbranded generator
+    5. Builds and packages the Azure generator locally
+    6. Updates the eng folder package.json artifacts in azure-sdk-for-net
+    7. Regenerates selected or all libraries using @azure-typespec/http-client-csharp and @typespec/http-client-csharp
+    8. Restores all modified artifacts to original state on success
+
+.PARAMETER AzureSdkForNetRepoPath
+    Required. The local file system path to the azure-sdk-for-net repository.
+
+.PARAMETER All
+    Optional. When specified, regenerates all libraries without prompting for selection.
+    If omitted, displays an interactive menu to select specific libraries to regenerate.
+
+.EXAMPLE
+    # Windows
+    .\RegenPreview-Local.ps1 -AzureSdkForNetRepoPath "C:\repos\azure-sdk-for-net"
+    
+    # Linux/macOS
+    ./RegenPreview-Local.ps1 -AzureSdkForNetRepoPath "/home/user/repos/azure-sdk-for-net"
+    
+    Prompts for library selection before regenerating.
+
+.EXAMPLE
+    .\RegenPreview-Local.ps1 -AzureSdkForNetRepoPath "C:\repos\azure-sdk-for-net" -All
```

Comment:
nit: probably can just name it RegenPreview unless we were thinking of having a separate script that we would use for pipeline runs?